src/perception/avt_341_perception_node.cpp

```diff
@@ -108,78 +104,48 @@ int main(int argc, char *argv[]) {
 
 	grid_created = false;
 
-	ros::init(argc, argv, "avt_341_perception_node");
-	
-	ros::NodeHandle n;
-	ros::Subscriber pc_sub = n.subscribe("avt_341/points",2,PointCloudCallback);
-	ros::Subscriber odom_sub = n.subscribe("avt_341/odometry",10, OdometryCallback);
-	ros::Publisher grid_pub = n.advertise<nav_msgs::OccupancyGrid>("avt_341/occupancy_grid", 1);
+	auto n = avt_341::node::init_node(argc, argv, "avt_341_perception_node");
+	auto pc_sub = n->create_subscription<avt_341::msg::PointCloud2>("avt_341/points",2,PointCloudCallback);
+    auto odom_sub = n->create_subscription<avt_341::msg::Odometry>("avt_341/odometry",10, OdometryCallback);
+    auto grid_pub = n->create_publisher<avt_341::msg::OccupancyGrid>("avt_341/occupancy_grid", 1);
 
-	if (ros::param::has("~grid_width") && ros::param::has("~grid_height")){
-		float grid_width, grid_height;
-		ros::param::get("~grid_width",grid_width);
-		ros::param::get("~grid_height",grid_height);
-		grid.SetSize(grid_width,grid_height);
-	}
-	else{
-		float grid_size = 500.0f;
-		if (ros::param::has("~grid_size")){
-    		ros::param::get("~grid_size",grid_size);
-  		}	
-		grid.SetSize(grid_size);
-	}
+    float grid_width, grid_height;
+    n->get_parameter("~grid_width", grid_width, 200.0f);
+    n->get_parameter("~grid_height", grid_height, 200.0f);
+    grid.SetSize(grid_width,grid_height);
 
+    float grid_res, grid_llx, grid_lly, warmup_time, thresh;
+    bool use_elevation;
 
-	float grid_res = 1.0f;
-	if (ros::param::has("~grid_res")){
-    	ros::param::get("~grid_res",grid_res);
-  	}	
-  	float grid_llx=-250.0f; 
-  	if (ros::param::has("~grid_llx")){
-    	ros::param::get("~grid_llx",grid_llx);
-  	}	
-  	float grid_lly=-250.0f; 
-  	if (ros::param::has("~grid_lly")){
-    	ros::param::get("~grid_lly",grid_lly);
- 	 }	
-	float warmup_time = 1.0f;
-	if (ros::param::has("~warmup_time")){
-    	ros::param::get("~warmup_time",warmup_time);
-  	}	
-	float thresh = 1.0f;
-	if (ros::param::has("~threshold")){
-		ros::param::get("~threshold",thresh);
-	}
-	bool use_elevation = false;
-	if (ros::param::has("~use_elevation")){
-		ros::param::get("~use_elevation",use_elevation);
-	}
-	if (ros::param::has("~use_registered")){
-		ros::param::get("~use_registered",use_registered);
-	}
-	if (ros::param::has("~overhead_clearance")){
-		ros::param::get("~overhead_clearance",overhead_clearance);
-	}
+    n->get_parameter("~grid_res", grid_res, 1.0f);
+    n->get_parameter("~grid_llx", grid_llx, -100.0f);
+    n->get_parameter("~grid_lly", grid_lly, -100.0f);
+
+    n->get_parameter("~warmup_time", warmup_time, 1.0f);
+    n->get_parameter("~slope_threshold", thresh, 1.0f);
```

Comment:
This used to be ~threshold in the previous version but I noticed that in the launch file it is defined as slope_threshold so I changed it to this.